components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/dao/impl/AuthenticatorManagementDAOImpl.java

```diff
@@ -0,0 +1,340 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.common.dao.impl;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.database.utils.jdbc.NamedPreparedStatement;
+import org.wso2.carbon.identity.application.common.constant.AuthenticatorMgtErrorConstants.ErrorMessages;
+import org.wso2.carbon.identity.application.common.constant.AuthenticatorMgtSQLConstants.Column;
+import org.wso2.carbon.identity.application.common.constant.AuthenticatorMgtSQLConstants.Query;
+import org.wso2.carbon.identity.application.common.dao.AuthenticatorManagementDAO;
+import org.wso2.carbon.identity.application.common.exception.AuthenticatorMgtException;
+import org.wso2.carbon.identity.application.common.exception.AuthenticatorMgtServerException;
+import org.wso2.carbon.identity.application.common.model.LocalAuthenticatorConfig;
+import org.wso2.carbon.identity.application.common.model.Property;
+import org.wso2.carbon.identity.application.common.model.VerificationAuthenticatorConfig;
+import org.wso2.carbon.identity.base.AuthenticatorPropertyConstants.AuthenticationType;
+import org.wso2.carbon.identity.base.AuthenticatorPropertyConstants.DefinedByType;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class implements the AuthenticatorManagementDAO interface which perform CRUD operation on database.
+ */
+public class AuthenticatorManagementDAOImpl implements AuthenticatorManagementDAO {
+
+    private static final Log LOG = LogFactory.getLog(AuthenticatorManagementDAOImpl.class);
+    public static final String IS_TRUE_VALUE = "1";
+    public static final String IS_FALSE_VALUE = "0";
+    public static final String LOCAL_IDP_NAME = "LOCAL";
+
+    public AuthenticatorManagementDAOImpl() {
+    }
+
+    @Override
+    public LocalAuthenticatorConfig addUserDefinedLocalAuthenticator(LocalAuthenticatorConfig authenticatorConfig,
+                Integer tenantId, AuthenticationType type) throws AuthenticatorMgtException {
+
+        Connection dbConnection = IdentityDatabaseUtil.getDBConnection(true);
+
+        try (NamedPreparedStatement statement = new NamedPreparedStatement(dbConnection, Query.ADD_AUTHENTICATOR_SQL)) {
+
+            statement.setString(Column.NAME, authenticatorConfig.getName());
+            statement.setString(Column.DISPLAY_NAME, authenticatorConfig.getDisplayName());
+            statement.setString(Column.DEFINED_BY, authenticatorConfig.getDefinedByType().toString());
+            statement.setString(Column.AUTHENTICATION_TYPE, type.toString());
+            int isEnabled = authenticatorConfig.isEnabled() ? 1 : 0;
```

Comment:
why we go with int ? why can't we go with boolean ? Is this the way it's implemented today ?

Reply:
This statement can be used as a direct argument for setInt function

Reply:
Addressed with commit d72ebd31b7150e7c562faf83ee80300a60e1e5d8